make/autoconf/jdk-version.m4

```diff
@@ -69,149 +69,163 @@ AC_DEFUN_ONCE([JDKVER_SETUP_JDK_VERSION_NUMBERS],
   AC_SUBST(JDK_RC_PLATFORM_NAME)
   AC_SUBST(HOTSPOT_VM_DISTRO)
 
+  # Note: UTIL_ARG_WITH treats empty strings as valid values when OPTIONAL is false!
+
   # Setup username (for use in adhoc version strings etc)
-  AC_ARG_WITH([build-user], [AS_HELP_STRING([--with-build-user],
-      [build username to use in version strings])])
-  if test "x$with_build_user" = xyes || test "x$with_build_user" = xno; then
-    AC_MSG_ERROR([--with-build-user must have a value])
-  elif test "x$with_build_user" != x; then
-    USERNAME="$with_build_user"
-  else
-    # Outer [ ] to quote m4.
-    [ USERNAME=`$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ]
-  fi
+  # Outer [ ] to quote m4.
+  UTIL_ARG_WITH(NAME: build-user, TYPE: string,
+    RESULT: USERNAME,
+    DEFAULT: [ `$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ],
```

Comment:
Can you move this outside the UTIL_ARG_WITH block and assign it to a variable? I'm worried this might be evaluated multiple times otherwise.

Also, I know this was not present before, but please add something along the lines of:
`DEFAULT_DESC: [current username, sanitized]`

Reply:
Will do, thanks